source/postprocess/visualization.cc

```diff
@@ -320,6 +329,16 @@ namespace aspect
           && (this->get_timestep_number() != 0))
         return std::pair<std::string,std::string>();
 
+      // up the counter of the number of the file by one, but not in
+      // the very first output step. if we run postprocessors on all
+      // iterations, only increase file number in the first nonlinear iteration
+      const bool first_output = (this->get_timestep_number() == 0)
```

Comment:
this logic causes some headache to understand. What about we initialize `output_file_number` in the constructor with `numbers::invalid_unsigned_int`, since it is the number of the last output file written. The if below would then look like the following:
```
if (output_file_number == numbers::invalid_unsigned_int)
  output_file_number = 0;
else if (increase_file_number)
  ++output_file_number;
```